simtools/model/array_model.py

```diff
@@ -1,19 +1,473 @@
+                return telName, dict()
+            else:
+                # Case 2: data has a wrong type
+                msg = 'ArrayConfig has wrong input for a telescope'
+                self._logger.error(msg)
+                raise InvalidArrayConfigData(msg)
+        # End of _proccessData
+
+        if telName in self._arrayConfigData.keys():
+            # Specific info for this telescope
+            return _proccessSingleTelescope(self._arrayConfigData[telName])
+        else:
+            # Grabing the default option
+            return _proccessSingleTelescope(self._arrayConfigData['default'][telSize])
+    # End of _getSingleTelescopeInfoFromArrayConfig
+
+    def printTelescopeList(self):
+        ''' Print out the list of telescopes for quick inspection. '''
+        for telData, telModel in zip(self.layout, self._telescopeModel):
+            print('Name: {}\t Model: {}'.format(telData.name, telModel.telescopeName))
+
+    def exportSimtelTelescopeConfigFiles(self):
+        '''
+        Export sim_telarray config files for all the telescopes
+        into the output model directory.
+        '''
+        exportedModels = list()
+        for telModel in self._telescopeModel:
+            name = (
+                telModel.telescopeName
+                + ('_' + telModel.extraLabel if telModel.extraLabel != '' else '')
+            )
+            if name not in exportedModels:
+                self._logger.debug('Exporting config file for tel {}'.format(name))
+                telModel.exportConfigFile()
+                exportedModels.append(name)
+            else:
+                self._logger.debug('Config file for tel {} already exists - skipping'.format(name))
+
+    def exportSimtelArrayConfigFile(self):
+        '''
+        Export sim_telarray config file for the array into the output model
+        directory.
+        '''
+
+        # Setting file name and the location
+        configFileName = names.simtelArrayConfigFileName(
+            self.layoutName,
+            self.site,
+            self.modelVersion,
+            self.label
+        )
+        self._configFilePath = self._configFileDirectory.joinpath(configFileName)
+
+        # Writing parameters to the file
+        self._logger.info('Writing array config file - {}'.format(self._configFilePath))
+        with open(self._configFilePath, 'w') as file:
+            header = (
+                '%{}\n'.format(50 * '=')
+                + '% ARRAY CONFIGURATION FILE\n'
+                + '% Site: {}\n'.format(self.site)
+                + '% LayoutName: {}\n'.format(self.layoutName)
+                + '% ModelVersion: {}\n'.format(self.modelVersion)
+                + ('% Label: {}\n'.format(self.label) if self.label is not None else '')
+                + '%{}\n\n'.format(50 * '=')
+            )
+            file.write(header)
+
+            # Be carefull with the formating - simtel is sensitive
```

Comment:
Super pedantic: do we call sim_telarray now always simtel?

Reply:
We have been doing it in some contexts. If you say that this is a problem, we can change that.

Reply:
Huge problem (no, just kidding - but be prepared the KB might complain at some point)